Source/WebCore/rendering/RenderTableSection.cpp

```diff
@@ -436,13 +437,18 @@ void RenderTableSection::distributeRemainingExtraLogicalHeight(LayoutUnit& extra
     if (extraLogicalHeight <= 0 || !m_rowPos[totalRows])
         return;
 
-    // FIXME: m_rowPos[totalRows] - m_rowPos[0] is the total rows' size.
-    LayoutUnit totalRowSize = m_rowPos[totalRows];
     LayoutUnit totalLogicalHeightAdded;
     LayoutUnit previousRowPosition = m_rowPos[0];
+    float totalRowSize = m_rowPos[totalRows] - previousRowPosition;
+    if (!totalRowSize) {
+        // It prevents division by '0' to avoid 'ASSERTION FAILED: !firstBody() || !extraLogicalHeight'.
+        extraLogicalHeight = 0;
+        return;
+    }
     for (unsigned r = 0; r < totalRows; r++) {
         // weight with the original height
-        totalLogicalHeightAdded += extraLogicalHeight * (m_rowPos[r + 1] - previousRowPosition) / totalRowSize;
+        float heightToAdd = extraLogicalHeight * (m_rowPos[r + 1] - previousRowPosition) / totalRowSize;
+        totalLogicalHeightAdded = std::min<float>(totalLogicalHeightAdded + std::ceil(heightToAdd), extraLogicalHeight);
```

Comment:
Using `std::ceil` seems like a bad idea. There’s nothing special about 1px, right? Do you know the answer @alanbaradlay, @anttijk or some other expert?

Reply:
I don't know the answer but mixing `LayoutUnit` (totalLogicalHeightAdded), `float` (std:min<float>) and an `integer` value (std::ceil) certainly needs some explanation here.